Reorganize "How to contribute" with new items, rewordings, and more cross-links (#44)
godotengine/godot-contributing-docs: organization/how_to_contribute.rst

.. - **Buy Official Merch**

Provide feedback
----------------

- **Report bugs & other issues**
  As active users of the engine, you are better equipped to identify bugs and other issues than anyone else.
  To let us know about your findings, fill out this `bug report form <https://github.com/godotengine/godot/issues/new/choose>`_ on our GitHub.
  Make sure to include as much information as possible to ensure these issues can easily be reproduced by others.

- **Propose and discuss ideas**
  To meaningfully improve Godot, we first need the community to tell us what needs to be improved.
  To get started, please visit our section on :ref:`ideas and change proposals <doc_contributing_ideas>`.

- **Test development versions**
  While it is recommended to use the stable releases for your projects, you can :ref:`help us test <doc_reporting_issues>`
  dev releases, betas, and release candidates by opening a copy of your project in them and checking what problems this

>   introduces or maybe even solves. Make sure to have a backup ready, since this can produce irreversible changes.

Reviewer: ```suggestion
  might introduce or maybe even solves. Make sure to have a backup ready, since using a pre-release can produce irreversible changes.
```
I think this sounds better by not implying it *will* cause problems, also I'd change the second part as it currently implies a backup might cause the changes, alternatively:
```suggestion
  might introduce or maybe even solves. Make sure to have a backup ready, to avoid irreversible changes.
```

  by discussing whether new features are useful and appropriate, and giving feedback on how to improve fixes or new features.

  You can start by browsing pull requests `directly on GitHub <https://github.com/godotengine/godot/pulls>`__,
  or by finding linked pull requests of `bug reports <https://github.com/godotengine/godot/issues>`__ and of
  `proposed changes <https://github.com/godotengine/godot-proposals/issues>`__.

Participate in Godot's development
----------------------------------

- **Engine code (mainly C++)**
  If you are a programmer, you can contribute to Godot by :ref:`working on its source code <doc_intro_to_engine_contributions>`.
  You can participate in almost every step of the :ref:`PR workflow <doc_pr_workflow>`! Most commonly, contributors
  fix bugs and make improvements by :ref:`opening pull requests <doc_creating_pull_requests>`.
  Once you start getting more confident with Godot's codebase, we'd also greatly appreciate help in the
  :ref:`code review process <doc_pr_review_guidelines>`!

  The engine development is coordinated by :ref:`teams <doc_areas>` and discussed on the `contributor RocketChat <https://chat.godotengine.org/>`__.

>   If you are serious about making PRs you should join us there!

Reviewer 2: We refer to it as "Godot Contributors Chat" elsewhere:

```suggestion
  The engine development is coordinated by :ref:`teams <doc_areas>` and discussed on the `Godot Contributors Chat <https://chat.godotengine.org/>`__.
  If you are serious about making PRs, you should join us there!
```

  :ref:`Contribute documentation <doc_documentation_overview>` to improve the Godot Engine with your technical writing skills.

- **Translations (spoken languages other than English)**
  Are you interested in making the Godot Engine more accessible to non-English speakers?
  Contribute to our :ref:`community translations <doc_editor_and_docs_localization>` on
  `Weblate <https://hosted.weblate.org/projects/godot-engine/godot/>`__.

- **Bugsquad & triage**
  With so many bug reports and pull requests being opened each day, the triage team — also called bugsquad — is
  invaluable to keep things organized.
  If you'd like to get involved, please visit :ref:`doc_bug_triage_intro`!

- **Contribute to other Godot resources**
  Besides the engine itself, you can contribute to many other aspects of Godot, such as the :ref:`website <doc_contributing_to_the_website>`,
  the :ref:`C++ GDExtension bindings <doc_godot_cpp>`, and our :ref:`development resources <doc_resources>`. Many of
  these non-engine resources are made with different technology from Godot itself, so they require people with
  different skill sets working on them!

>   To find resources to contribute to, browse the navigation bar on the left (on mobile, click the menu button at the top

Reviewer: ```suggestion
  To find resources to contribute to, browse the navigation bar on the left (on mobile, click the menu button on the top
```
